policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -425,7 +442,89 @@ impl ServiceRoutes {
                 policy.opaque = self.opaque;
                 modified = true;
             }
+            if self.accrual != policy.accrual {
+                policy.accrual = self.accrual;
+                modified = true;
+            }
             modified
         });
     }
 }
+
+fn parse_accrual_config(
+    annotations: &std::collections::BTreeMap<String, String>,
+) -> Result<Option<FailureAccrual>> {
+    annotations
+        .get("balancer.linkerd.io/failure-accrual")
+        .map(|mode| {
+            if mode == "consecutive" {
+                let max_failures = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-max-failures")
+                    .map(|s| s.parse::<u32>())
+                    .transpose()?
+                    .unwrap_or(7);
+
+                let max_penalty = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-max-penalty")
+                    .map(|s| parse_duration(s))
+                    .transpose()?
+                    .unwrap_or_else(|| time::Duration::from_secs(60));
+
+                let min_penalty = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-min-penalty")
+                    .map(|s| parse_duration(s))
+                    .transpose()?
+                    .unwrap_or_else(|| time::Duration::from_secs(1));
+                let jitter = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-jitter-ratio")
+                    .map(|s| s.parse::<f32>())
+                    .transpose()?
+                    .unwrap_or(0.5);
+                if min_penalty > max_penalty {
+                    bail!(
+                        "min_penalty ({min_penalty:?}) cannot exceed max_penalty ({max_penalty:?})",
+                    );
+                }
+
+                Ok(FailureAccrual::Consecutive {
+                    max_failures,
+                    backoff: Backoff {
+                        min_penalty,
+                        max_penalty,
+                        jitter,
+                    },
+                })
+            } else {
+                bail!("unsupported failure accrual mode: {mode}");
+            }
+        })
+        .transpose()
+}
+
+//TODO: check what we do in proxy for this.
```

Comment:
The proxy has a similar function: https://github.com/linkerd/linkerd2-proxy/blob/main/linkerd/app/src/env.rs#L890

Reply:
I looked into it. The regex approach is smaller and reads well. There shouldn't be a functional difference between the two. I wanted to see how hard it would be to add it in, and we have to bring in a few additional functions that the proxy counterpart relies on (e.g `parse` and `parse_number`). Not sure it's worth doing.

I removed the comment for now.